cmd/security-secrets-setup/edgex-stack-security.yml

```diff
@@ -0,0 +1,560 @@
+#  * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+#  * or implied. See the License for the specific language governing permissions and limitations under
+#  * the License.
+#  *
+#  * @author: Jim White, Dell
+#  * EdgeX Foundry, Edinburgh, version 1.0.1
+#  * added: July 26, 2019
+#  *******************************************************************************/
+
+version: '3'
+volumes:
+  db-data:
+  log-data:
+  consul-config:
+  consul-data:
+  portainer_data:
+  vault-config:
+  vault-file:
+  vault-logs:
+
+services:
+  volume:
+    image: edgexfoundry/docker-edgex-volume:1.0.0
+    container_name: edgex-files
+    networks:
+      edgex-network:
+        aliases:
+            - edgex-files
+    volumes:
+      - db-data:/data/db
+      - log-data:/edgex/logs
+      - consul-config:/consul/config
+      - consul-data:/consul/data
+
+  consul:
+    image: consul:1.3.1
+    ports:
+      - "8400:8400"
+      - "8500:8500"
+      - "8600:8600"
+    container_name: edgex-core-consul
+    hostname: edgex-core-consul
+    networks:
+      edgex-network:
+        aliases:
+            - edgex-core-consul
+    volumes:
+      - db-data:/data/db
+      - log-data:/edgex/logs
+      - consul-config:/consul/config
+      - consul-data:/consul/data
+    depends_on:
+      - volume
+
+  config-seed:
+    image: edgexfoundry/docker-core-config-seed-go:1.0.0
+    container_name: edgex-config-seed
+    hostname: edgex-core-config-seed
+    networks:
+      edgex-network:
+        aliases:
+            - edgex-core-config-seed
+    volumes:
+      - db-data:/data/db
+      - log-data:/edgex/logs
+      - consul-config:/consul/config
+      - consul-data:/consul/data
+    depends_on:
+      - volume
+      - consul
+
+  vault:
+    image: vault:1.2.3
+    container_name: edgex-vault
+    hostname: edgex-vault
+    networks:
+      edgex-network:
+        aliases:
+            - edgex-vault
+    ports:
+      - "8200:8200"
+    cap_add:
+      - "IPC_LOCK"
+    command: '/usr/bin/dumb-init -- /bin/sh -c "while test \! -f /run/edgex/secrets/edgex-vault/.security-secrets-setup.complete; do sleep 1; done; exec  /usr/local/bin/docker-entrypoint.sh server -log-level=info"'
+    environment:
+      - 'VAULT_ADDR=https://edgex-vault:8200'
+      - 'VAULT_LOCAL_CONFIG=listener "tcp" {   address = "edgex-vault:8200"   tls_disable = "0"   cluster_address = "edgex-vault:8201"  tls_min_version = "tls12"  tls_client_ca_file ="/vault/config/pki/EdgeXFoundryCA/EdgeXFoundryCA.pem"  tls_cert_file ="/run/edgex/secrets/edgex-vault/server.crt"  tls_key_file = "/run/edgex/secrets/edgex-vault/server.key"  tls_perfer_server_cipher_suites = true }  backend "consul" { path = "vault/" address = "edgex-core-consul:8500" scheme = "http" redirect_addr = "https://edgex-vault:8200" cluster_addr = "https://edgex-vault:8201" }  default_lease_ttl = "168h"  max_lease_ttl = "720h"'
```

Comment:
any chance to avoid this long string???

Reply:
Inject the file from docker-edgex-volume as suggested for the shell script, then add the `-config`  parameter pointing at the file when launching vault.